### PR TITLE
Muhammed shihebi/issue23

### LINKED ISSUE #23: Get /events

### PR DESCRIPTION
**Pull request by muhammed-shihebi**
This branch implements the GET /api/event/ endpoint with all of its tests and swagger documentation

### CODE REVIEW

File: src/utility/utils.js [hunk 1]
```diff
@@ -1,7 +1,41 @@
...
+        mockEvents.push({
+            publisherId: mongoose.Types.ObjectId(),
+            title: `Event ${i}`,
+            content: `Event ${i} content`,
+            coverImage:
+                'https://images.unsplash.com/photo-1518791841217-8f162f1e1131?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=800&q=60',
+            date: moment('2022-01-01', 'YYYY-MM-DD')
+                .add(i - 1, 'days')
+                .format('YYYY-MM-DD'), // adding i days to the current date
+            categories: variables.CATEGORIES.slice(i - 1, i + 1),
```
Comment (ramaalshaban) [inline src/utility/utils.js:23]: what if the i+1in slice function is bigger than the number of categories provided in the variables/categories file?

File: src/utility/utils.js [hunk 2]
```diff
@@ -1,7 +1,41 @@
...
+            address: {
+                city: variables.CITIES[Math.floor(i / 2)],
+                country: 'Turkey',
```
Comment (ramaalshaban) [inline src/utility/utils.js:26]: Should we add a condition to check the cities variable numbers that is provided in the variables/categories file?

File: src/models/event.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ const mongoose = require('mongoose');
 const variables = require('../utility/variables');
 
 const eventSchema = mongoose.Schema({
-    puplisherId: {
+    publisherId: {
```
Comment (ramaalshaban) [inline src/models/event.js:5]: it took me one hour of my life to discover this typo! 
I CRY

**Review by ramaalshaban — APPROVED** [review]
Aferin SHEHABIII!

**Review by ramaalshaban — DISMISSED** [review]
a comment!

File: src/models/event.js (hunk 3 above)
Comment (muhammed-shihebi) [reply src/models/event.js:5]: Kanka,  I CRY too 😭

File: src/controllers/events.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+const { addDummyEventData } = require('../utility/utils');
+
+async function getEvents(req, res) {
+    // await addDummyEventData();
+    try {
+        // between filter options there is an implicit AND operator
+        const filter = {};
+
+        // gets the event if one of the categories passed is in the event
+        if (req.query.categories) {
```
Comment (halitbatur) [inline src/controllers/events.js]: can we destruct all these values and then check them?

File: src/controllers/events.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        // Which fields to populate from the user model
+        const requiredUserField = [
+            'id',
+            'firstName',
+            'lastName',
+            'email',
+            'profileImage',
+        ];
+
+        // due to the validation, we are sure that "from" and "to" are defined togheter
```
Comment (halitbatur) [inline src/controllers/events.js]: typo

**Review by halitbatur — APPROVED** [review]
Small changes and its ready to be merged

**Review by sarahmood — APPROVED** [review]

**Pull request merged**